### PR TITLE
update(date-arithmetic): version 4 updates

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- simplify module definition
- add missing and new methods
- correct 'seconds' constant
- amend tests
- add maintainer

https://github.com/jquense/date-math/releases/tag/v4.1.0

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## date-arithmetic ([<kbd>unpkg</kbd>](https://unpkg.com/browse/date-arithmetic@latest/))
was missing the following properties:
1. diff
1. inRange
1. max
1. min
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e8e18ca873cd39fa0f1c16f4ccad1677f76b8ee9
</p>

File: types/date-arithmetic/index.d.ts [hunk 1]
```diff
@@ -1,38 +1,77 @@
...
-    const dateArithmetic: dateArithmeticStatic;
-    export = dateArithmetic;
-}
+/** Compare two dates and return true if date is less than date2 */
+export function lt(date: Date, date2: Date): boolean;
+
+/** Compare two dates and return true if date is less or equal than date2 */
+export function lte(date: Date, date2: Date): boolean;
+
+export type StartOfWeeek = 0 | 1 | 2 | 3 | 4 | 5 | 6;
```
Comment (sandersn) [inline types/date-arithmetic/index.d.ts:77]:
```suggestion
export type StartOfWeek = 0 | 1 | 2 | 3 | 4 | 5 | 6;
```

**Review by sandersn — CHANGES_REQUESTED** [review]
Found a typo!

File: types/date-arithmetic/index.d.ts [hunk 2]
```diff
@@ -1,38 +1,77 @@
...
+export function minutes(date: Date): number;
+export function minutes(date: Date, value: number): Date;
+export function hours(date: Date): number;
+export function hours(date: Date, value: number): Date;
+export function date(date: Date): number;
+export function date(date: Date, value: number): Date;
+export function day(date: Date): number;
+export function day(date: Date, value: number): Date;
+export function weekday(date: Date, value: Date): number;
+export function weekday(date: Date, value: Date, firstOfWeek: StartOfWeeek): Date;
```
Comment (peterblazejewicz) [inline types/date-arithmetic/index.d.ts:32]:
```suggestion
export function weekday(date: Date, value: Date, firstOfWeek: StartOfWeek): Date;
```

File: types/date-arithmetic/index.d.ts [hunk 3]
```diff
@@ -1,38 +1,77 @@
...
-    eq(date: Date, date2: Date): Boolean;
+/**
+ * The opposite of `startOf`
+ */
+export function endOf(date: Date, unit: Unit): Date;
+/**
+ * Return a new date with the relevent date parts zero'd out.
+ * You only need to provide a firstOfWeek when the unit is 'week'
+ */
+export function startOf(date: Date, unit: 'week', firtOfWeek: StartOfWeeek): Date;
```
Comment (peterblazejewicz) [inline types/date-arithmetic/index.d.ts:53]:
```suggestion
export function startOf(date: Date, unit: 'week', firtOfWeek: StartOfWeek): Date;
```

File: types/date-arithmetic/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/date-arithmetic/index.d.ts]: Ouch, fixed, amended in other places,
thx!

**Review by sandersn — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Ready to merge 💋

**Pull request merged**